### ISSUE TITLE
Auto indents don't work in match blocks

### ISSUE THREAD

**Issue by NathanLovato** [GDQuest/emacs-gdscript-mode#29]
Match blocks are a special case when it comes to indents, as they have nested blocks with the patterns, e.g. below, `Events.UpgradeChoices.HEALTH:`:

```gdscript
func _on_Upgrade_Choice_made(choice: int) -> void:
	match choice:
		Events.UpgradeChoices.HEALTH:
			stats.add_modifier(max_health, {value=15})
			_health = stats.get_max_health()
			health_bar.max_value = max_health
			health_bar.value = _health
```

We indent on keywords like match, func, etc., or at the end of an if block, but we need a rule to indent after the first block and colon inside a match block.

I'm not sure how to get this one. We need to indent the block after the `:$` regex, after the match keyword. There can be lots of stuff in these lines, including var assignments and more.

**Comment by NathanLovato** [comment]
@ofrank123 I'm assigning you to this if you can tackle it

**Comment by ofrank123** [comment]
Working on it, is there any time where we wouldn't want to start a new block if after the `:$` regex?  If not it may be best to use that as the block-start regex instead of using the keywords.   Otherwise I can work on a programmatic solution, but it may take me a while because the indentation code is so dense, and I'd need to make a `gdscript-is-block-start` function and replace the `looking-at` lines with this, which I could see getting fairly messy.

**Comment by NathanLovato** [comment]
> is there any time where we wouldn't want to start a new block if after the :$ regex?

I can't think of any, this seems good to me. If it's simple to implement, we might as well try it and see if any bug arises.

**Comment by NathanLovato** [comment]
Closed by b028badcd93c16e8be53ab9eab74524847b1c4f0
